presto-main/src/test/java/io/prestosql/operator/TestDynamicFilterSourceOperator.java

```diff
@@ -383,48 +386,71 @@ public void testCollectEmptyBuildSide()
                 ImmutableList.of(TupleDomain.withColumnDomains(ImmutableMap.of(
                         new DynamicFilterId("0"),
                         Domain.create(
-                                ValueSet.ofRanges(range(BIGINT, 0L, true, (long) maxPositionsCount, true)),
+                                ValueSet.ofRanges(range(BIGINT, 0L, true, (long) maxDistinctValues, true)),
                                 false)))));
     }
 
     @Test
-    public void testMultipleColumnsCollectMinMaxRangeWhenTooManyPositions()
+    public void testMultipleColumnsCollectBelowDistinctValuesLimit()
     {
-        int maxPositionsCount = 300;
+        int maxDistinctValues = 101;
         Page largePage = new Page(
                 createLongSequenceBlock(0, 101),
-                createColorSequenceBlock(100, 201),
+                createLongSequenceBlock(100, 201),
                 createLongSequenceBlock(200, 301));
 
+        List<TupleDomain<DynamicFilterId>> expectedTupleDomains = ImmutableList.of(
+                TupleDomain.withColumnDomains(ImmutableMap.of(
+                        new DynamicFilterId("0"), Domain.create(
+                                ValueSet.copyOf(BIGINT, LongStream.rangeClosed(0L, 100L).boxed().collect(toImmutableList())),
+                                false),
+                        new DynamicFilterId("1"), Domain.create(
+                                ValueSet.copyOf(BIGINT, LongStream.rangeClosed(100L, 200L).boxed().collect(toImmutableList())),
+                                false),
+                        new DynamicFilterId("2"), Domain.create(
+                                ValueSet.copyOf(BIGINT, LongStream.rangeClosed(200L, 300L).boxed().collect(toImmutableList())),
+                                false))));
+        assertDynamicFilters(maxDistinctValues, ImmutableList.of(BIGINT, BIGINT, BIGINT), ImmutableList.of(largePage), expectedTupleDomains);
+    }
+
+    @Test
+    public void testMultipleColumnsCollectMinMaxRangeWhenTooManyDistinctValues()
+    {
+        int maxDistinctValues = 100;
+        Page largePage = new Page(
+                createLongSequenceBlock(0, 101),
+                createColorRepeatBlock(100, 101),
+                createLongRepeatBlock(200, 101));
+
         List<TupleDomain<DynamicFilterId>> expectedTupleDomains = ImmutableList.of(
                 TupleDomain.withColumnDomains(ImmutableMap.of(
                         new DynamicFilterId("0"), Domain.create(ValueSet.ofRanges(
                                 range(BIGINT, 0L, true, 100L, true)), false),
                         new DynamicFilterId("2"), Domain.create(ValueSet.ofRanges(
-                                range(BIGINT, 200L, true, 300L, true)), false))));
-        assertDynamicFilters(maxPositionsCount, ImmutableList.of(BIGINT, COLOR, BIGINT), ImmutableList.of(largePage), expectedTupleDomains);
+                                equal(BIGINT, 200L)), false))));
```

Comment:
why is it `equal`?

Reply:
Used repeat block for this column to verify case of many distinct values only on one column